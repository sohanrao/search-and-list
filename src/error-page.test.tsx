import { RouterProvider, createMemoryRouter } from "react-router-dom";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import ErrorPage from "./error-page";
import userEvent from "@testing-library/user-event";
import Landing from "./routes/landing";

describe("Error Page", () => {
  it("renders the Error Page component", async () => {
    const FAKE_EVENT = { name: "test event" };
    const routes = [
      {
        path: "/",
        errorElement: <ErrorPage />,
        element: <Landing />,
        loader: () => FAKE_EVENT,
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ["/foo"],
      initialIndex: 1,
    });

    const { container } = render(<RouterProvider router={router} />);

    expect(container).toMatchSnapshot();

    const p = screen.findByText("Sorry, an unexpected error has occurred.");
    const btnHome = screen.getByText("Home");
    expect(p).to.exist;
    expect(btnHome).to.exist;

    await userEvent.click(btnHome);

    expect(screen.getByText("All You Can Read")).to.exist;
  });
});
