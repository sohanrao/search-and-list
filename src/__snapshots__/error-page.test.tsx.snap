// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error Page > renders the Error Page component 1`] = `
<div>
  <div
    id="error-page"
  >
    <h1>
      Oops!
    </h1>
    <p>
      Sorry, an unexpected error has occurred.
    </p>
    <p>
      <i>
        Not Found
      </i>
    </p>
    <a
      href="/"
    >
      Home
    </a>
  </div>
</div>
`;
